File: 0953-verifying-an-alien-dictionary/0953-verifying-an-alien-dictionary.java
class Solution {
    public boolean isAlienSorted(String[] words, String order){
        HashMap<Character, Integer> orderMap = new HashMap<>();

        for (int index = 0; index < order.length(); index++) {
            orderMap.put(order.charAt(index), index);
        }

        for (int ind = 0; ind < words.length - 1; ind++) {
            for (int innerInd = 0; innerInd < words[ind].length(); innerInd++) {
                if(innerInd >= words[ind+1].length()){
                    return false;
                }
                if(words[ind].charAt(innerInd) != words[ind+1].charAt(innerInd)){
                    int currentWordLetterOrder = orderMap.get(words[ind].charAt(innerInd));
                    int nextWordLetterOrder = orderMap.get(words[ind+1].charAt(innerInd));
                    
                    if(nextWordLetterOrder < currentWordLetterOrder){
                        return false;
                    }else{
                        break;
                    }
                }
            }
        }
        return true;
    }
}

File: 0953-verifying-an-alien-dictionary/0953-verifying-an-alien-dictionary.py
class Solution:
    def isAlienSorted(self, words: List, order: str) -> bool:
     
        order_dict = {}


        for index in range(len(order)):
            order_dict[order[index]] = order_dict.get(order[index], index+1)

        for ind in range(0, len(words)-1):
            current_sorted = True
            for inner_ind in range(len(words[ind])):
                if inner_ind >= len(words[ind+1]):
                    return False
                if order_dict[words[ind+1][inner_ind]] > order_dict[words[ind][inner_ind]]:
                    current_sorted = True
                    break
                elif order_dict[words[ind+1][inner_ind]] < order_dict[words[ind][inner_ind]]:
                    return False
                
        return True